src/components/table-hoc/TableHeaderWithSort.tsx

```diff
@@ -0,0 +1,79 @@
+import * as classNames from 'classnames';
+import * as React from 'react';
+import * as _ from 'underscore';
+import {IReactVaporState} from '../../ReactVapor';
+import {IDispatch, ReduxConnect} from '../../utils/ReduxUtils';
+import {Svg} from '../svg/Svg';
+import {TableHeaderActions} from './actions/TableHeaderActions';
+import {ITableWithSortState} from './reducers/TableWithSortReducers';
+
+export interface ITableHeaderWithSortOwnProps {
+    id: string;
+    tableId: string;
+    isDefault?: boolean;
+}
+
+export interface ITableHeaderStateProps {
+    sorted: boolean;
+}
+
+export interface ITableHeaderWithSortDispatchProps {
+    onMount: () => void;
+    onSort: () => void;
+    onUnmount: () => void;
+}
+
+export interface ITableHeaderWithSortProps extends
+    ITableHeaderWithSortOwnProps,
+    Partial<ITableHeaderStateProps>,
+    Partial<ITableHeaderWithSortDispatchProps> {}
+
+const mapStateToProps = (state: IReactVaporState, ownProps: ITableHeaderWithSortOwnProps) => {
+    const tableSort: ITableWithSortState = _.findWhere(state.tableHOCHeader, {id: ownProps.id});
+
+    return {
+        sorted: tableSort && tableSort.isAsc,
+    };
+};
+
+const mapDispatchToProps = (
+    dispatch: IDispatch,
+    ownProps: ITableHeaderWithSortOwnProps,
+): ITableHeaderWithSortDispatchProps => ({
+    onMount: () => dispatch(TableHeaderActions.addTableHeader(ownProps.id, ownProps.tableId, ownProps.isDefault)),
+    onSort: () => dispatch(TableHeaderActions.sortTable(ownProps.id)),
+    onUnmount: () => dispatch(TableHeaderActions.removeTableHeader(ownProps.id)),
+});
+
+@ReduxConnect(mapStateToProps, mapDispatchToProps)
+export class TableHeaderWithSort extends React.Component<ITableHeaderWithSortProps & React.HTMLAttributes<HTMLTableHeaderCellElement>> {
+    private handleClick(e: React.MouseEvent<HTMLTableHeaderCellElement>) {
+        this.props.onSort();
+    }
+
+    componentDidMount() {
+        this.props.onMount();
+    }
+
+    componentWillUnmount() {
+        this.props.onUnmount();
+    }
+
+    render() {
+        const headerCellClasses = classNames(this.props.className, 'admin-sort', {
+            'admin-sort-ascending': this.props.sorted === true,
+            'admin-sort-descending': this.props.sorted === false,

```

Comment:
```
true --> sort ascending
false --> sort descending  
undefined --> unsorted  
```  
?

Reply:
If we want to continue with the same logic has we have now. Its the right think to do. IMO, if we have a sortable table, It better to sort asc or desc on a specific row on the first render. In this case, the unsorted was removed and we can set the right sort svg on sorted column.  It's look weird to have an initial state only available on the first render and not available after that.  @Charles-ErickBG I don't know the right answer but I think the UX can help to figure out what kind of behaviour we want in the new table.

Reply:
@MichelML Yes, undefined is unsorted (for this column)

@pochretien there's a prop on the header that you need to set (isDefault) to make this column the default sort. This way you control which column is the default one

Reply:
Good, but its possible to have the default column sorted with undefined in the first render ? or we need to set asc or desc

Reply:
Talked live, everything is fine like it is